sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -74,10 +74,20 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   }
 
   /**
-   * @return All metrics containing metrics of this SparkPlan.
+   * @return Executor side metrics of this SparkPlan.
    */
   def metrics: Map[String, SQLMetric] = Map.empty
 
+  /**
+   * @return Driver side metrics of this SparkPlan.
+   */
+  @transient def driverMetrics: Map[String, SQLMetric] = Map.empty
```

Comment:
does `@transient` work for methods?

Reply:
ah, doesn't work, just need to add @trasient in the subclass.